### DISCUSSION TITLE
Sweex PP110

### DISCUSSION THREAD

**Discussion by tomasenskede** [hassio-addons/app-nut#303] [General]
I am trying to connect Home Assistant with my Sweex PP100 UPS.

I managed to connect from debian with driver richcomm_usb

In HA I tested with both the default driver usbhid-ups aswell as the richcomm_usb

My conifg;
![image](https://github.com/hassio-addons/addon-nut/assets/32595351/136fc370-1220-46c1-9249-87c31ea09b96)

My Proxmox conifg of the USB-port/connection:
![image](https://github.com/hassio-addons/addon-nut/assets/32595351/5f4bd2ad-db08-4448-b28f-ec9383f619bf)

My log;
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service base-addon-banner: starting
-----------------------------------------------------------
 Add-on: Network UPS Tools
 Manage battery backup (UPS) devices
-----------------------------------------------------------
 Add-on version: 0.12.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 10.2  (amd64 / qemux86-64)
 Home Assistant Core: 2023.6.1
 Home Assistant Supervisor: 2023.06.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-timezone: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
[21:50:40] INFO: Configuring timezone (Europe/Stockholm)...
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service base-addon-timezone successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/nut.sh
[21:50:41] INFO: Setting mode to netserver...
[21:50:41] INFO: Generating /etc/nut/upsd.users...
[21:50:41] FATAL: 
[21:50:41] FATAL: Setting a username is required!
[21:50:41] FATAL: 
[21:50:41] FATAL: Please username in the 'users[0].username' option.
[21:50:41] FATAL: 
[21:50:41] FATAL: If unsure, check the add-on manual for more information.
[21:50:41] FATAL: 
cont-init: info: /etc/cont-init.d/nut.sh exited 1
cont-init: info: running /etc/cont-init.d/nutclient.sh
cont-init: info: /etc/cont-init.d/nutclient.sh exited 0
cont-init: warning: some scripts exited nonzero
s6-rc: warning: unable to start service legacy-cont-init: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
/run/s6/basedir/scripts/rc.init: fatal: stopping the container.
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-timezone: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-timezone successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by frenck** [comment]
The solution is in the logs...

```
[21:50:41] FATAL: Setting a username is required!
[21:50:41] FATAL: 
[21:50:41] FATAL: Please username in the 'users[0].username' option.
```

../Frenck

**Reply by tomasenskede** [reply]
All ok now?

Network UPS Tools - UPS driver controller 2.8.0
cont-init: info: /etc/cont-init.d/nut.sh exited 0
cont-init: info: running /etc/cont-init.d/nutclient.sh
cont-init: info: /etc/cont-init.d/nutclient.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun upsd (no readiness notification)
services-up: info: copying legacy longrun upsmon (no readiness notification)
s6-rc: info: service legacy-services successfully started
[11:20:59] INFO: Starting the UPS information server...
   0.000000	fopen /run/nut/upsd.pid: No such file or directory
   0.000498	Could not find PID file '/run/nut/upsd.pid' to see if previous upsd instance is already running!
   0.001382	[D1] debug level is '1'
   0.001746	listening on 0.0.0.0 port 3493
   0.002490	Connected to UPS [PP100]: richcomm_usb-PP100
   0.005618	Running as foreground process, not saving a PID file
[11:21:00] INFO: Starting the UPS monitor and shutdown controller...
   0.000000	fopen /run/nut/upsmon.pid: No such file or directory
   0.000598	Could not find PID file to see if previous upsmon instance is already running!
   0.001008	Using power down flag file /etc/killpower
   0.001310	UPS: PP100@localhost (primary) (power value 1)
   0.001559	[D1] debug level is '1'
   0.001871	[D1] debug level is '1'
   0.002088	Warning: running as one big root process by request (upsmon -p)
   0.002371	[D1] Saving PID 460 into /run/nut/upsmon.pid
Init SSL without certificate database
   0.015444	[D1] Trying to connect to UPS [PP100@localhost]
   0.524672	User upsmonmaster@127.0.0.1 logged into UPS [PP100]
   0.017269	[D1] Logged into UPS PP100@localhost

**Reply by tomasenskede** [reply]
0.017678	User upsmonmaster@127.0.0.1 logged into UPS [PP100]
   0.006281	[D1] Logged into UPS PP100@localhost
{"message":"Event nut.ups_event fired."}Network UPS Tools upsmon 2.8.0 <--- this is alarm from UPS, power off
{"message":"Event nut.ups_event fired."}Network UPS Tools upsmon 2.8.0 <--- power on
{"message":"Event nut.ups_event fired."}Network UPS Tools upsmon 2.8.0 <-- power off
{"message":"Event nut.ups_event fired."}Network UPS Tools upsmon 2.8.0 <--- power on

But I can't get the client to connect to this server... to handle the alerts....  

See my other post on this issue:
https://github.com/networkupstools/nut/issues/1966#issuecomment-1593004781

**Reply by tomasenskede** [reply]
The case has been resolved. NUT Server has been successfully installed on a fresh Debian virtual server, and the NUT Client has been connected to the NUT server. Everything is now functioning as expected.
![image](https://github.com/hassio-addons/addon-nut/assets/32595351/5c689478-3374-4bf6-ae8f-6655db629b73)
![image](https://github.com/hassio-addons/addon-nut/assets/32595351/d2ef52ce-c4ac-4b16-840e-0e271425bad6)
